From c2e8cc6b1f1a22227d475e145de9fa847a4c407d Mon Sep 17 00:00:00 2001
Message-Id: <c2e8cc6b1f1a22227d475e145de9fa847a4c407d.1548789435.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 29 Jan 2019 20:07:03 +0100
Subject: [PATCH] EogScrollView: Set scale=1 for the image surface

Rendering the pixbuf to a surface with a different scale will apply
filtering, which we don't want at this step.

This fixes zoomed-in images being blurry even when anti-aliasing has
been disabled.
---
 src/eog-scroll-view.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/eog-scroll-view.c b/src/eog-scroll-view.c
index 76ca8d49..93eb1c51 100644
--- a/src/eog-scroll-view.c
+++ b/src/eog-scroll-view.c
@@ -211,6 +211,7 @@ create_surface_from_pixbuf (EogScrollView *view, GdkPixbuf *pixbuf)
 	                                             CAIRO_CONTENT_COLOR | CAIRO_CONTENT_ALPHA,
 	                                             gdk_pixbuf_get_width (pixbuf),
 	                                             gdk_pixbuf_get_height (pixbuf));
+	cairo_surface_set_device_scale (surface, 1.0, 1.0);
 	cr = cairo_create (surface);
 	gdk_cairo_set_source_pixbuf (cr, pixbuf, 0, 0);
 	cairo_paint (cr);
-- 
2.20.1

